### ISSUE TITLE
Headless browser for fetching proof post

### ISSUE THREAD

**Issue by nykma** [nextdotid/proof_server#39]
tl;dr: add a headless browser mechanism alongside current API-based proof post fetching channel.

## Motivation

For now, `proof_server` performs `proof_post` fetching by API or single HTTP GET. But in real world, there may be no way for some `platform`s to do this way. Including:

- No APIs at all provided by `platform`
- API needs each user to do an `OAuth` to fetch their own data
- API has very strict quota and will finally exceed if `proof_service` users grows.
- User data cannot be fetched by single `HTTP GET` (e.g. JavaScript-driven `XHR` is performed in that page)
- HTML DOM struct is changed frequently, can't define a fixed selector to query the data from DOM tree.
- Remote HTTP server has a strict anti-bot mechanism, single `HTTP GET` will be blocked in a short amount of time.

To make `proof_server` supports more and more services later, we need a headless browser to simulate normal user's page-browsing flow.

## Design goal

- The headless browser module can be kind of independent from current `proof_server`'s code structure. With a simple wrapper, `proof_server` can communicate with it in different processes.
- The headless browser module will find user's proof post using single `regexp`, instead of parsing and `css-style` selecting DOM.

**Comment by ef3n9r98** [comment]
Hi @nykma, I know some devs who'd be interested in working on this. Is there a budget for fixing this issue, and if so, how much? 

I'd put this issue up on [OpenQ](http://www.openq.dev/).

Thanks!

**Comment by nykma** [comment]
Sorry but this issue is part of a Gitcoin bounty event. We've received some good contributions towards this and should have closed this issue. Will close now. 

Sorry again.... :cry:
